operator/pkg/util/cert/util.go

```diff
@@ -0,0 +1,246 @@
+	}
+	rootCert, err := ioutil.ReadFile(filepath.Join(certDir, RootCertName))
+	if err != nil {
+		log.Warnf("can not read %q from %s", RootCertName, certDir)
+		return err
+	}
+	cakey, err := ioutil.ReadFile(filepath.Join(certDir, CAKeyName))
+	if err != nil {
+		log.Warnf("can not read %q from %s", CAKeyName, certDir)
+		return err
+	}
+	caCert, err := ioutil.ReadFile(filepath.Join(certDir, CACertName))
+	if err != nil {
+		log.Warnf("can not read %q from %s", CACertName, certDir)
+		return err
+	}
+	certChain, err := ioutil.ReadFile(filepath.Join(certDir, CertChainName))
+	if err != nil {
+		log.Warnf("can not read %q from %s", CertChainName, certDir)
+		return err
+	}
+	err = createNamespace(client, ns, dryRun)
+	if err != nil {
+		return err
+	}
+	err = createSecret(client, ns, rootCert, cakey, caCert, certChain, dryRun)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func GenerateCertsAndKey(client kubernetes.Interface, ns string, dryRun bool, outputDir string) error {
+	_, scrtErr := client.CoreV1().Secrets(ns).Get(PlugInSecret, metav1.GetOptions{})
+	if scrtErr == nil {
+		return nil
+	}
+	org := "istio.io"
+	options := util.CertOptions{
+		TTL:          cmd.DefaultSelfSignedCACertTTL,
+		Org:          org,
+		IsCA:         true,
+		IsSelfSigned: true,
+		RSAKeySize:   caKeySize,
+		IsDualUse:    true,
+	}
+	rootCertBytes, rootKeyBytes, ckErr := util.GenCertKeyFromOptions(options)
+	if ckErr != nil {
+		return ckErr
+	}
+
+	rootCert, err := util.ParsePemEncodedCertificate(rootCertBytes)
+	if err != nil {
+		return err
+	}
+
+	rootKey, err := util.ParsePemEncodedKey(rootKeyBytes)
+	if err != nil {
+		return err
+	}
+
+	opts := util.CertOptions{
+		Org:          org + "intermediate",
+		TTL:          cmd.DefaultSelfSignedCACertTTL,
+		SignerCert:   rootCert,
+		SignerPriv:   rootKey,
+		IsCA:         true,
+		IsSelfSigned: false,
+		RSAKeySize:   caKeySize,
+	}
+
+	intermidiateCertBytes, intermidiateKeyBytes, ckErr := util.GenCertKeyFromOptions(opts)
+	if ckErr != nil {
+		return ckErr
+	}
+	err = createNamespace(client, ns, dryRun)
+	if err != nil {
+		return err
+	}
+	err = createSecret(client, ns, rootCertBytes, intermidiateKeyBytes, intermidiateCertBytes, intermidiateCertBytes, dryRun)
+	if err != nil {
+		return err
+	}
+	opts.Org = org + ".intermediate1"
+	intermidiateCertBytes1, intermidiateKeyBytes1, ckErr := util.GenCertKeyFromOptions(opts)
+	if ckErr != nil {
+		return ckErr
+	}
+	opts.Org = org + ".intermediate2"
+	intermidiateCertBytes2, intermidiateKeyBytes2, ckErr := util.GenCertKeyFromOptions(opts)
+	if ckErr != nil {
+		return ckErr
+	}
+	err = saveCreds(outputDir, rootCertBytes, intermidiateCertBytes1, intermidiateKeyBytes1, intermidiateCertBytes2, intermidiateKeyBytes2)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func saveCreds(certDir string, rootCerts []byte, certPem1 []byte, privPem1 []byte, certPem2 []byte, privPem2 []byte) error {
```

Comment:
I would prefer to create a directory per cluster - this will also be needed in MC cases, and we must stop using the same cert in all clusters. 

@ayj

Reply:
fixed, put into 3 separate directories, one for rootcert, the other for the two intermediate ca